assets/robotest/upgrade-base/app.yaml

```diff
@@ -0,0 +1,158 @@
+apiVersion: cluster.gravitational.io/v2
+kind: Cluster
+metadata:
+  name: robotest
```

Comment:
What do we need two robotest apps for, "upgrade-base" and "current"? To be able to test application-only upgrades when runtime doesn't change?

Reply:
We need two (or more) separate manifests for a couple reasons:

1) Old versions of gravity do not support all the manifest directives that current versions do.  For master (this PR) we test upgrades from:
```
6.2.0 6.2.5 6.3.18 7.0.0 7.0.12 7.0.13 7.0.16 (this last one floats)
```
In particular, 6.2.x doesn't support `storage/openebs`.  We can make the entirely reasonable argument that we don't need to test 6.2 for this branch.  However the mechanism is important because this code will be backported and we do need to test upgrades from 5.5.x in 7.0.x.  And there are manifest differences between 5.5..x and 7.1.x.  We need them to test contemporary features.  This is why I alluded to upgrade testing helping with backports yesterday.

2) In order to exercise the upgrade hook, we *want* the upgrade base application to be different than the current application. Ideally the hooks for the new version will exercise both gravity and application functionality that wasn't previously available. We're not there yet, but that is where I'm aiming.

To flip your question: Why don't we have more upgrade from versions?

Considering the list from earlier:
```
6.2.0 6.2.5 6.3.18 7.0.0 7.0.12 7.0.13 7.0.16 (this last one floats)
```

We could very well have one base image for 6.2.x and a different one for 6.3 and 7.0.x. Or a different version of the application for each upgrade from version, supporting exactly what that gravity version supports.

There are a couple difficulties here:

1) Maintaining a separate application for each upgrade base version is tedious. It means each manifest change is multiplied by 5-10 edits.  There is potential to put together some code to only abstract & maintain the differences.  But this seemed complex for a v1.  I'm not looking to build out yaml/manifest templating.

2) There is no good way to keep a manifest current for the floating version. It'd require that each time you cut a release, you (or some automation) cut a PR to each branch that could be upgraded to/from.  This is not worth it at this point.

2) Makefiles are insufficient to elegantly handle this.  The pattern rules I've written are already byzantine and difficult to understand, without having many-to-one or one-to-one manifest-to-upgrade-images.  This would be considerably easier using go & @knisbet's work in #1917 but I can't afford to tie my work to his -- it'd  delay availability by weeks-to-months and make ports... challenging.

Reply:
Just to chime in on the topic of yaml templating - if it's ever a thing, I think [cue](github.com/cuelang/cue/) would be perfect for the job.